app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,70 @@
+class TweetsController < ApplicationController
+
+
+    get '/' do
+        erb :index
+    end
+    
+    get '/tweets' do
+        if logged_in?
+            erb :'tweets/tweets'   
+        else
+            redirect '/login'
+        end
+    end
+    
+    get '/tweets/new' do
+        if logged_in?
+            erb :"tweets/create_tweet"
+        else
+            redirect '/login'
+        end
+    end
+    
+    get '/tweets/:id' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params["id"])
+            erb :"tweets/show_tweet"
+        else
+            redirect "/login"
+        end
+    end
+    
+    get '/tweets/:id/edit' do
+        if logged_in?
+            @tweet = Tweet.find_by_id(params["id"])
+            erb :'tweets/edit_tweet'
+        else
+            redirect '/login'
+        end
+    end
+
+    patch '/tweets/:id' do
+        @tweet = Tweet.find_by_id(params["id"])
+        if logged_in? && !params["content"].empty?
+            @tweet.update(:content => params["content"])
+            redirect "/tweets/#{@tweet.id}/edit"
+        else
+            redirect "/tweets/#{@tweet.id}/edit"
+        end        
+    end
+
+    post '/tweets' do
+        if logged_in? && !params["content"].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks in your model more easily.